$MirOS$

	Even “faked” this spews link-time warnings for *all* users.
	Remove this insecure nonsense. We could do better and make
	it always use arc4random for seeding, but first to update…

--- cipher/random.c.orig	Tue Jan  8 11:28:16 2008
+++ cipher/random.c	Wed May 28 21:51:04 2014
@@ -1338,22 +1338,11 @@ gather_faked (void (*add)(const void*, s
     {
       log_info(_("WARNING: using insecure random number generator!!\n"));
       initialized=1;
-#ifdef HAVE_RAND
-      srand( time(NULL)*getpid());
-#else
-      srandom( time(NULL)*getpid());
-#endif
     }
 
   p = buffer = gcry_xmalloc( length );
   n = length;
-#ifdef HAVE_RAND
-  while ( n-- )
-    *p++ = ((unsigned)(1 + (int) (256.0*rand()/(RAND_MAX+1.0)))-1);
-#else
-  while ( n-- )
-    *p++ = ((unsigned)(1 + (int) (256.0*random()/(RAND_MAX+1.0)))-1);
-#endif
+  arc4random_buf(p, n);
   add_randomness ( buffer, length, origin );
   gcry_free (buffer);
   return 0; /* okay */
